src/ReactiveSocket.h

```diff
@@ -107,6 +107,8 @@ class ReactiveSocket {
       std::unique_ptr<DuplexConnection> newConnection,
       const ResumeIdentificationToken& token);
 
+  const DuplexConnection& getDuplexConnection();
```

Comment:
please change this to pointer instead of a reference return type. When the RS instance is disconnected it doesn't have DuplexConnection associated.